qemu/tests/block_kill_reconnect_with_remote_storage.py

```diff
@@ -0,0 +1,157 @@
+        error_context.context(
+            "Run IO test when in reconnecting loop", logging.info)
+        for iteration in range(self.repeat_times):
+            error_context.context("Wait %s seconds" % self.reconnect_time_wait,
+                                  logging.info)
+            time.sleep(self.reconnect_time_wait)
+            self.run_io_test("test_file")
+
+    def check_data_disk_resume(self, test_disk):
+        error_context.context(
+            "check data disk resumed", logging.info)
+        for iteration in range(self.repeat_times):
+            logging.info("Wait %s seconds" % self.reconnect_time_wait)
+            time.sleep(self.reconnect_time_wait)
+            self.run_io_test(test_disk)
+
+    def clean_images(self, net_status):
+        """
+        recover nbd image access
+        """
+        if net_status:
+            self.resume_net_with_iptables()
+        # stop nbd image export
+        self.stop_export_local_image_with_nbd()
+
+        super(BlockReconnectTest, self).clean_images()
+
+    def stop_export_local_image_with_nbd(self):
+        logging.info("Stop export nbd data disk image.")
+        self.nbd_export.stop_export()
+
+    def test_run(self):
+        disk_storage_name = self.get_disk_storage_name(
+            self.params.get("find_system_disk_cmd"),
+            self.params.get("find_data_disk_cmd"))
+        data_disk = disk_storage_name[1]
+        if self.disk_op_cmd:
+            self.run_io_test(data_disk)
+        self.stop_export_local_image_with_nbd()
+        self.break_net_with_iptables()
+        self.reconnect_loop_io()
+        self.nbd_export.export_image()
+        self.resume_net_with_iptables()
+        self.check_data_disk_resume(data_disk)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test kill/reconnect of block devices.
+
+        1) Boot up guest with a data disk.
+        2) Do I/O on data disk
+        3) Kill nbd export image process.
+        4) Apply a firewall on nbd data disk's port
+        5) Check guest system disk function
+        6) Drop the firewall
+        7) Export the nbd data disk image
+        8) Do I/O on data disk
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    reconnect_test = BlockReconnectTest(test, params, env)
+    reconnect_test.disk_op_cmd = params.get("disk_op_cmd")
+    reconnect_test.disk_op_timeout = int(params.get("disk_op_timeout", 360))
+    reconnect_test.timeout = int(params.get("login_timeout", 360))
+    reconnect_test.nbd_port_data1 = params.get("nbd_port_data1")
+    reconnect_test.repeat_times = int(params.get("repeat_times"))
+    reconnect_test.reconnect_time_wait = int(params.get("reconnect_time_wait"))
+    reconnect_test.prepare_test()
+    reconnect_test.vm = env.get_vm(params["main_vm"])
+    reconnect_test.vm.verify_alive()
+    reconnect_test.test_run()
```

Comment:
refer to above, here is reconnect_test.run_test() and you can replace your test_run with do_test